life_expectancy/transformation_interface.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+import pandas as pd
+from life_expectancy.loading_strategies import Region
+
+
+class Transformation(ABC):
+    @abstractmethod
+    def transform(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Transform the input data and return the transformed data."""
+
+    def is_necessary(self, data: pd.DataFrame) -> bool:
+        """Test if transformation is necessry"""
+        return True
+
+
+class TransformationPipeline:
+    def __init__(self):
+        self.transformations = []
+
+    def add_transformation(self, transformation: Transformation):
```

Comment:
does this typing make sense? are your others transformation classes transformation type as well?

Reply:
The typing is intended to ensure that only objects that inherit from the `Transformation` abstract base class can be added to the pipeline. All transformation classes in the codebase are designed to inherit from this base class, so the typing should be appropriate.